crates/extensions/c8y_mapper_ext/src/availability/builder.rs

```diff
@@ -32,19 +31,20 @@ pub struct AvailabilityBuilder {
 impl AvailabilityBuilder {
     pub fn new(
         config: AvailabilityConfig,
-        mqtt: &mut (impl MessageSource<MqttMessage, Vec<ChannelFilter>> + MessageSink<PublishMessage>),
+        mqtt_in: &mut impl MessageSource<MqttMessage, Vec<ChannelFilter>>,
+        mqtt_out: &mut impl MessageSink<MqttMessage>,
```

Comment:
Nice. You put the finger on the root of the issue.

The badly name `mqtt` connector is not a true MQTT connector (`impl MessageSource<MqttMessage, TopicFilter> + MessageSink<MqttMessage>`),  *but* the c8y mapper forwarding some of the messages to this actor.

You broke the loop (between this actor and the c8y actor), by sending the response messages directly to an MQTT actor. I wonder if we can go a step further. Simply disconnect this actor from the c8y actor and let it consume its input directly from an MQTT actor.

```suggestion
        mqtt: &mut (impl MessageSource<MqttMessage, TopicFilter> + MessageSink<MqttMessage>)
```

You will have not only to adapt `AvailabilityBuilder::mqtt_message_builder()` (exactly as you currently propose) but also the `AvailabilityBuilder::channels()` function but not the tests. Construction of the mapper will have to be changed but lightly.